spec/report_spec.rb

```diff
@@ -58,6 +58,46 @@
       expect(report).to be_an_instance_of(described_class)
       expect(report.id).to match('trkrep')
     end
+
+    it 'creates a report with custom columns and additional columns' do
+      report = described_class.create(
+        start_date: Fixture.report_start_date,
+        end_date: Fixture.report_end_date,
+        type: 'shipment',
+        columns: ['usps_zone'],
+        additional_columns: %w[from_name from_company],
+      )
+
+      expect(report).to be_an_instance_of(described_class)
+      expect(report.id).to match('shprep')
+      expect(report.columns).to eq(['usps_zone'])
+      expect(report.additional_columns).to eq(%w[from_name from_company])
+
+      retrieved_report = described_class.retrieve(report.id)
+
+      expect(retrieved_report).to be_an_instance_of(described_class)
+      expect(retrieved_report.start_date).to eq(report.start_date)
+      expect(retrieved_report.end_date).to eq(report.end_date)
```

Comment:
Unsure what the purpose of retrieving the report is, we just want to make sure creating it works as expected.

Reply:
Resolved in 838f54d0dbb44f5b78595d4a0840983b7d997834